crmd/notify.c

```diff
@@ -20,120 +20,501 @@
 #include <crm/crm.h>
 #include <crm/msg_xml.h>
 #include "notify.h"
+#include "crmd_messages.h"
 
 char *notify_script = NULL;
 char *notify_target = NULL;
+GListPtr notify_list = NULL;
+
+typedef struct {
+    char *id;
+    char *path;
+    int timeout;
+    char *tstamp_format;
+    char *info;
```

Comment:
For the first suggestion, I was thinking that the script itself would stay the same on all hosts, but someone could write it such that it has its own configuration file, which could vary by host. I think that's more useful than a single info parameter.

The second suggestion would let the entire configuration stay within pacemaker.

I can see some value in per-recipient timeouts: the error messages would say which one timed out, and if the number of recipients varied over time, the user wouldn't have to worry about changing the timeout. But that's a relatively small benefit. I'd be open to keeping recipient separate from instance_attributes, but it does seem cleaner and more consistent to keep it in (if going that route). For example, we already have logic for selecting instance attribute sets based on rules, so it would be a little easier to enable rules for alerts in the future, compared to a completely new XML object.